libs/net/kafka/KafkaListener`.cs

```diff
@@ -20,6 +21,7 @@ public class KafkaListener<TKey, TValue> : IKafkaListener<TKey, TValue>, IDispos
     private bool _disposed = false;
     private bool _open = false;
     private ConsumeResult<TKey, TValue>? _currentResult = null;
+    private readonly ConcurrentDictionary<ConsumeResult<TKey, TValue>, ConsumeResult<TKey, TValue>> _currentResults = new();
```

Comment:
I don't think it makes much sense to have a dictionary that has both `key = value`.

Reply:
Updated, thanks